<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <meta name="author" content="Sergio Peignier">
  <title>Sergio-Peignier-Home</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="./css/style.css">
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  
  <script>
  /* When the user clicks on the button,
  toggle between hiding and showing the dropdown content */
  function myFunctionDN() {
      document.getElementById("dropnotes").classList.toggle("show");
  }
  function myFunctionDT() {
      document.getElementById("dropteaching").classList.toggle("show");
  }
  // Close the dropdown if the user clicks outside of it
  window.onclick = function(e) {
    if (!e.target.matches('.dropbtn')) {
  
      var dropdowns = document.getElementsByClassName("dropdown-content");
      for (var d = 0; d < dropdowns.length; d++) {
        var openDropdown = dropdowns[d];
        if (openDropdown.classList.contains('show')) {
          openDropdown.classList.remove('show');
        }
      }
    }
  }
  </script>
  <header>
    <div class="menutop">
    <ul>
    <li><a href="./index.html">About</a></li>
    <li><a href="./CV.html">CV</a></li>
    <li><a href="./research.html">Research</a></li>
    <li><a href="./software.html">Software</a></th>
    <li class="dropdown">
      <a href="javascript:void(0)" class="dropbtn" onclick="myFunctionDT()">Teaching</a>
      <div class="dropdown-content" id="dropteaching">
        <a href="./python.html">Python</a>
        <a href="./Databases.html">Data Bases</a>
      </div>
    </li>
    <li class="dropdown">
      <a href="javascript:void(0)" class="dropbtn" onclick="myFunctionDN()">Notes</a>
      <div class="dropdown-content" id="dropnotes">
        <a href="./writing_scientific_papers.html">Some guidelines to write scientific papers</a>
      </div>
    </li>
    </ul>
    </div>
    </header>
</head>
<body>
<h1 id="software-and-side-projects">Software and side projects</h1>
<h2 id="recent-software-developed">Recent software developed</h2>
<h3 id="chameleoclust">Chameleoclust+</h3>
<p>ChameleoClust+ is a bio-inspired algorithm implementing an evolvable genome structure, including several bio-like features such as a variable genome length, both functional and non-functional elements and mutation operators including chromosomal rearrangements. The main purpose of the design of ChameleoClust+ is to take advantage of the large degree of freedom provided by its evolvable structure to detect various number of clusters in subspaces of various dimensions.</p>
<p>Chameleoclust+ runs as a python module implemented internally in C++. The software can be downloaded in the webpage of the <a href="http://evoevo.liris.cnrs.fr/chameleoclust/">EvoEvo Project</a>.</p>
<h3 id="nayesdog-rss-reader-with-naive-bayes-powered-recommendations">NayesDog: RSS reader with Naive Bayes powered recommendations</h3>
<p>Have You Ever Felt Like You Were Drowning in an endless sea of data while you try to retrieve valuable information? <a href="https://github.com/MLdog/nayesdog">NayesDog</a> is a recent project developed with <a href="https://iprokin.github.io/index.html">Ilya Prokin</a> that aims to provide a smart and helpful <a href="https://en.wikipedia.org/wiki/RSS">RSS</a> reader. As for most RSS readers, users can subscribe to their favorite feeds by entering the feed’s URI and then the reader inspects and shows the chosen feeds when required. However, unlike most readers, NayesDog reads the entries and sorts them according to your preferences before showing them. Therefore users may focus on news that have been classified as more interesting for them by the algorithm. NayesDog first converts feed entries into bag of words and then uses a Naive Bayes classifier to correlate words with the user preferences. Indeed, preference buttons (like / dislike) allows users to give a feedback to the algorithm regarding their preferences. The algorithm has been designed to adapt to evolvable preferences of users.</p>
<h3 id="time-series-analysis">Time Series Analysis</h3>
<p><a href="https://timeseriesanalysis.github.io/about_us.html">TimeSeriesAnalysis</a> is recent project created with <a href="https://github.com/bdenoun">Brice Denoun</a>. The aim of this project is to extract knowledge from EEG signals captured by low-cost sensors using deep learning, machine learning and time series analysis in order to prevent for example seizures and cerebrovascular accident dynamically. Our purpose is to make the software as useful, light and portable as possible so it could produce suitable results while being executed in a small device (e.g., a raspberry pi).</p>
<h3 id="adventurous-social-science">Adventurous Social Science</h3>
<p><a href="https://github.com/AdventurousSocialScience">Adventurous Social Science</a> is a recent project we started some months ago with <a href="https://www.google.fr/url?sa=t&amp;rct=j&amp;q=&amp;esrc=s&amp;source=web&amp;cd=8&amp;cad=rja&amp;uact=8&amp;ved=0ahUKEwiNtZKhxP7PAhUSahoKHV4aDOUQFgg4MAc&amp;url=https%3A%2F%2Fmx.linkedin.com%2Fin%2Fmois%25C3%25A9s-arizpe-rojo-05640b99&amp;usg=AFQjCNHw4CDnIzotIwSAADhwBiaI952-aQ&amp;sig2=_PIR36AemgjiQ2Q95cgvCw&amp;bvm=bv.136811127,d.d2s">Moises Arizpe Rojo</a>. The name of this project describe it well, we try to address social science questions but in a more adventurous and less conventional way. Our first goal is always to build an interesting question we will address. Then we proceed to collect data from social networks (e.g., through Twitter API) and we apply state-of-the-art machine learning techniques to understand better the collected observations. Finally we analyze our results and compare them to stablished theory from Social Sciences in order to give a possible answer to our initial problem (e.g. <a href="https://medium.com/adventurous-social-science">most recent posts</a>).</p>
<h3 id="kernels-for-graph-comparisons">Kernels for graph comparisons</h3>
<p>The development of new drugs is a expensive and slow procedure. An important step in new molecules development is the “test” phase. This particularly slow and expensive step is significantly accelerated and cheapened if Structure-Activity Relationship Analysis (SAR) is applied to the tested chemical compounds. We studied here how to predict molecular activity with graph Kernels. Kernels4graphs implements several kernel based methods for classification of graphs. A basic SVM algorithm and 4 different graph Kernels implementations (Nth order walk kernel, Geometric walk kernel, Markovian random walk kernel and Subtree kernel) are provided. The different approaches have been applied to Structure-Activity Relationship Analysis (SAR) to predict the molecular activity of differente chemical compounds described as labeled graphs. This project can be found in its github <a href="https://github.com/SergioPeignier/kernels4graphs">repository</a>.</p>
<h3 id="secondary-structure-rna-prediction">Secondary structure RNA prediction</h3>
</body>
</html>
